### PR TITLE
Add a security section to the book

### PR DESCRIPTION
**Pull request by michaelsproul**
## Proposed Changes

This PR adds a section on security to the book which is intended for security researchers. I've also uploaded the 2020 security assessment by Trail of Bits which we have not previously shared publicly (but have permission from both ToB and SigP to do so now).

### CODE REVIEW

**Review by paulhauner — APPROVED** [review]

File: book/src/security.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Security
+========
```
Comment (chong-he) [inline book/src/security.md]: I think we need a header `#` here and a blank line after the header to make Markdown linter happy

```suggestion
# Security

```

File: book/src/security.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Security
+
+
```
Comment (chong-he) [inline book/src/security.md]: I accidentally deleted the separating line `======`. this should do it.

```suggestion
========
```

Comment (dapplion) [reply book/src/security.md]: I think it was me! 🙏

**Comment by michaelsproul** [comment]
@mergify queue

**Pull request merged**